index.js

```diff
@@ -141,44 +181,59 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+let lineNumber = 0
+function takeATicketNumber(line){
+  // lineLength = line.length +1 
+  lineNumber++
+  line.push(lineNumber)
+  return "Welcome you are ticket number " + lineNumber
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log("%c----------", "color: red") 
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
+
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc() {
+   
+}
+
```

Comment:
This ones a tough one! Try watching this scrimba if you haven't already and try again: [https://scrimba.com/scrim/cLgEmZUZ](https://scrimba.com/scrim/cLgEmZUZ)

In Scrimba, you can pause the video and literally start playing around with the code in the editor from the video